app/models/rent.rb

```diff
@@ -0,0 +1,4 @@
+class Rent < ApplicationRecord
+  validates :user, :book, :begin, :end, presence: true
```

Comment:
the card does not ask for it but with your knowledge so far you should be able to add a custom validation saying that the `begin`  date should be before the `end` date
When you are modeling stuff you should always think about this kind of logic